src/Symfony/Bundle/FrameworkBundle/Tests/Command/RouterDebugCommandTest.php

```diff
@@ -0,0 +1,148 @@
+        );
+    }
+
+    public function testWithSortByNameParameter()
+    {
+        $tester = $this->createCommandTester();
+        $ret = $tester->execute(['--sort' => 'name'], ['decorated' => false]);
+
+        $this->assertEquals(0, $ret, 'Returns 0 in case of success');
+        $this->assertRegExp(
+            '/(alpha.*?)(.\s)*(bravo.*?)(.\s)*(charlie.*?).*/mi',
+            $tester->getDisplay()
+        );
+        $this->assertRegExp(
+            '/(\/second.*?)(.)*\s(.)*(\/second.*?)(.)*\s(.)*(\/first.*?)(.)*\s(.)*/mi',
+            $tester->getDisplay()
+        );
+    }
+
+    public function testWithSortByPathParameter()
+    {
+        $tester = $this->createCommandTester();
+        $ret = $tester->execute(['--sort' => 'path'], ['sort' => 'path', 'decorated' => false]);
+
+        $this->assertEquals(0, $ret, 'Returns 0 in case of success');
+        $this->assertRegExp(
+            '/(charlie.*?)(.\s)*(alpha.*?)(.\s)*(bravo.*?).*/mi',
+            $tester->getDisplay()
+        );
+        $this->assertRegExp(
+            '/(\/first.*?)(.)*\s(.)*(\/second.*?)(.)*\s(.)*(\/second.*?)(.)*\s(.)*/mi',
+            $tester->getDisplay()
+        );
+    }
+
+    public function testWithSortByPathParameterDoesNotMissAliases()
+    {
+        $tester = $this->createCommandTester();
+        $ret = $tester->execute(['--sort' => 'path'], ['sort' => 'path', 'decorated' => false]);
+
+        $this->assertEquals(0, $ret, 'Returns 0 in case of success');
+        $this->assertRegExp(
+            '/(charlie.*?)(.\s)*(alpha.*?)(.\s)*(bravo.*?).*/mi',
+            $tester->getDisplay()
+        );
+        $this->assertRegExp(
+            '/(\/first.*?)(.)*\s(.)*(\/second.*?)(.)*\s(.)*(\/second.*?)(.)*\s(.)*/mi',
+            $tester->getDisplay()
+        );
+    }
+
+    private function createCommandTester(): CommandTester
+    {
+        $application = new Application($this->getKernel());
+        $application->add(new RouterDebugCommand($this->getRouter()));
+
+        return new CommandTester($application->find('debug:router'));
+    }
+
+    private function getRouter()
+    {
+        $routeCollection = new RouteCollection();
+        $routeCollection->add('charlie', new Route('first'));
+        $routeCollection->add('alpha', new Route('second'));
+        $routeCollection->add('bravo', new Route('second'));
+        $requestContext = new RequestContext();
+        $router = $this->getMockBuilder('Symfony\Component\Routing\RouterInterface')->getMock();
+        $router
+            ->expects($this->any())
+            ->method('getRouteCollection')
+            ->willReturn($routeCollection);
+        $router
+            ->expects($this->any())
+            ->method('getContext')
+            ->willReturn($requestContext);
+
+        return $router;
+    }
+
+    private function getKernel()
+    {
+        $container = $this->getMockBuilder('Symfony\Component\DependencyInjection\ContainerInterface')->getMock();
```

Comment:
```suggestion
        $container = $this->createMock('Symfony\Component\DependencyInjection\ContainerInterface');
```

and can you use a `class` constant here?

Reply:
Do you mean declaring `$container` as `const CONTAINER`?